Minio/Helper/RequestUtil.cs

```diff
@@ -0,0 +1,161 @@
+﻿
+/*
+ * Minio .NET Library for Amazon S3 Compatible Cloud Storage, (C) 2017 Minio, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+using Minio.Exceptions;
+using Minio.Helper;
+using System;
+using System.Text.RegularExpressions;
+
+namespace Minio
+{
+    public class RequestUtil
+    {
+        public static Uri GetEndpointURL(string endPoint, bool secure)
+        {
+            if (endPoint.Contains(":"))
+            {
+                string[] parts = endPoint.Split(':');
+                string host = parts[0];
+                string port = parts[1]; 
+                if (!s3utils.IsValidIP(host) && !s3utils.IsValidDomain(host))
+                {
+                    throw new InvalidEndpointException("Endpoint: " + endPoint + " does not follow ip address or domain name standards.");
+                }
+
+            } 
+            else
+            {
+                if (!s3utils.IsValidIP(endPoint) && !s3utils.IsValidDomain(endPoint))
+                {
+                    throw new InvalidEndpointException("Endpoint: " + endPoint + " does not follow ip address or domain name standards.");
+                }
+            }
+
+            Uri uri = TryCreateUri(endPoint, secure);
+            RequestUtil.ValidateEndpoint(uri,endPoint);
+            return uri;
+
+        }
+     
+
+        public static Uri MakeTargetURL(string endPoint, bool secure, string bucketName = null, string region = null, bool usePathStyle = true)
+        {
+            // For Amazon S3 endpoint, try to fetch location based endpoint.
+            string host = endPoint;
+            if (s3utils.IsAmazonEndPoint(endPoint))
+            {
+                // Fetch new host based on the bucket location.
+                host = AWSS3Endpoints.Instance.endpoint(region);
+                if (!usePathStyle)
+                {
+                    host = utils.UrlEncode(bucketName) + "." + utils.UrlEncode(host) + "/";
+                }
+            }
+            Uri uri = TryCreateUri(host,secure);
+            return uri;
+        }
+
+        public static Uri TryCreateUri(string endpoint,bool secure)
```

Comment:
TryCreate gives an indication that you may not give a valid Uri back - the system function is named likewise